object_tracker/object_tracker.go

```diff
@@ -190,6 +191,16 @@ func (t *myTracker) run(stream gostream.VideoStream, cancelableCtx context.Conte
 			return
 		default:
 			start := time.Now()
+
+			// Reset if it's the exact time or it's been more than 24 hours since last reset
+			itsBeen := start.Sub(t.lastReset)
+			h, m, s := start.Hour(), start.Minute(), start.Second()
+			if itsBeen >= 24*time.Hour ||
+				(h == t.lastReset.Hour() && m == t.lastReset.Minute() && s == t.lastReset.Second()) {
+				t.classCounter = make(map[string]int)
```

Comment:
Since this is running in a go routine, you want to make sure about race conditions that might appear. One thing to really prevent this would be to consider using a `sync.Map` object instead of a simple map.

Reply:
https://pkg.go.dev/sync#Map